### ISSUE TITLE
missing destination name \"d.path\" in *clickhouse.Disk"

### ISSUE THREAD

**Issue by armanzhakypbayev** [Altinity/clickhouse-backup#1007]
ClickHouse server version 23.10.3.5 (official build).
root@db-clickhouse-ipsdev2:~# clickhouse-backup --version
Version:         2.6.0
Git Commit:      b7faa4a9c273ece943e406ebcca663065f1c41db
Build Date:      2024-08-29

root@db-clickhouse-ipsdev2:~# clickhouse-backup create test --env CLICKHOUSE_PASSWORD=***
failed with:

2024-09-10 09:57:18.479 INF pkg/clickhouse/clickhouse.go:1071 > SELECT countIf(name='type') AS is_disk_type_present, countIf(name='object_storage_type') AS is_object_storage_type_present, countIf(name='free_space') AS is_free_space_present, countIf(name='disks') AS is_storage_policy_present FROM system.columns WHERE database='system' AND table IN ('disks','storage_policies')
2024-09-10 09:57:18.499 INF pkg/clickhouse/clickhouse.go:1071 > SELECT d.path, any(d.name) AS name, any(d.type) AS type, min(d.free_space) AS free_space, groupUniqArray(s.policy_name) AS storage_policies FROM system.disks AS d  LEFT JOIN (SELECT policy_name, arrayJoin(disks) AS disk FROM system.storage_policies) AS s ON s.disk = d.name GROUP BY d.path
2024-09-10 09:57:18.502 INF pkg/clickhouse/clickhouse.go:319 > clickhouse connection closed
2024-09-10 09:57:18.502 FTL cmd/clickhouse-backup/main.go:658 > error="clickhouse [ScanStruct]: missing destination name \"d.path\" in *clickhouse.Disk"

Can you please help me?

**Comment by Slach** [comment]
sorry for late reply, could you share result of

`clickhouse-client --password=*** -q "SELECT d.path, any(d.name) AS name, any(d.type) AS type, min(d.free_space) AS free_space, groupUniqArray(s.policy_name) AS storage_policies FROM system.disks AS d LEFT JOIN (SELECT policy_name, arrayJoin(disks) AS disk FROM system.storage_policies) AS s ON s.disk = d.name GROUP BY d.path FORMAT Vertical"`

**Comment by armanzhakypbayev** [comment]
Hi, thank you for your reply!

root@db-clickhouse-ipsdev2:~# clickhouse-client --password=*** -q "SELECT d.path, any(d.name) AS name, any(d.ty
pe) AS type, min(d.free_space) AS free_space, groupUniqArray(s.policy_name) AS storage_policies FROM system.disks AS d L
EFT JOIN (SELECT policy_name, arrayJoin(disks) AS disk FROM system.storage_policies) AS s ON s.disk = d.name GROUP BY d.
path FORMAT Vertical"
Row 1:
──────
d.path:           /var/lib/clickhouse/
name:             default
type:             local
free_space:       83931381760
storage_policies: ['default']

**Comment by Slach** [comment]
Oh, looks weird

I tried reproduce query
https://fiddle.clickhouse.com/f97f3db4-7188-42f0-9fdd-533e32077b00

for the same clickhouse-server version
but got different result

```
Row 1:
──────
path:             /var/lib/clickhouse/
name:             default
type:             local
free_space:       1985749671936
storage_policies: ['default']
```

I don't see `d.path`

could you share?

```sql
SELECT * FROM system.settings WHERE changed
```

**Comment by armanzhakypbayev** [comment]
yeah I see, actually I dont know did we change this settings but from the select it seems not

![image](https://github.com/user-attachments/assets/8097025c-ab77-4ddf-8ac9-ebc55f572805)

**Comment by armanzhakypbayev** [comment]
Also I have another clickhouse server:

~# clickhouse-client --user=*** --password=*** -q "SELECT d.path, any(d
.name) AS name, any(d.type) AS type, min(d.free_space) AS free_space, groupUniqArray(s.policy_name) AS storage_policies
FROM system.disks AS d LEFT JOIN (SELECT policy_name, arrayJoin(disks) AS disk FROM system.storage_policies) AS s ON s.d
isk = d.name GROUP BY d.path FORMAT Vertical"
Row 1:
──────
d.path:           /var/lib/clickhouse/
name:             default
type:             local
free_space:       63939837952
storage_policies: ['default']

~# clickhouse-server --version
ClickHouse server version 23.7.4.5 (official build).

changed setting in the second server the same except there is no 'max_query_size' (it is default)

**Comment by Slach** [comment]
could you share 
`clickhouse-client -q "SELECT * FROM system.settings WHERE changed FORMAT Vertical"`
as text

**Comment by armanzhakypbayev** [comment]
sure
```
Row 1:
──────
name:        max_query_size
value:       2621440
changed:     1
description: The maximum number of bytes of a query string parsed by the SQL parser. Data in the VALUES clause of INSERT queries is processed by a separate stream parser (that consumes O(1) RAM) and not affected by this restriction.
min:         ᴺᵁᴸᴸ
max:         ᴺᵁᴸᴸ
readonly:    0
type:        UInt64
default:     262144
alias_for:
is_obsolete: 0

Row 2:
──────
name:        join_use_nulls
value:       1
changed:     1
description: Use NULLs for non-joined rows of outer JOINs for types that can be inside Nullable. If false, use default value of corresponding columns data type.
min:         ᴺᵁᴸᴸ
max:         ᴺᵁᴸᴸ
readonly:    0
type:        Bool
default:     0
alias_for:
is_obsolete: 0

Row 3:
──────
name:        allow_experimental_analyzer
value:       1
changed:     1
description: Allow experimental analyzer
min:         ᴺᵁᴸᴸ
max:         ᴺᵁᴸᴸ
readonly:    0
type:        Bool
default:     0
alias_for:
is_obsolete: 0
```

**Comment by Slach** [comment]
wrong behavior fixed in clickhouse-server 24.3
https://fiddle.clickhouse.com/310de429-201b-4d4f-a7a0-e7549cf462e5

**Comment by Slach** [comment]
ok. found root reason
combination of join_use_nulls=1,  allow_experimental_analyzer=1
reproduced in 23.3+

fixed in 24.3
https://fiddle.clickhouse.com/310de429-201b-4d4f-a7a0-e7549cf462e5
